vslib/src/SwitchStateBaseMACsec.cpp

```diff
@@ -541,15 +541,15 @@ sai_status_t SwitchStateBase::loadMACsecAttrFromMACsecSA(
 
     macsecAttr.m_an = attr->value.u8;
 
-    SAI_METADATA_GET_ATTR_BY_ID(attr, SAI_MACSEC_SA_ATTR_ENCRYPTION_ENABLE, attrCount, attrList);
+    SAI_METADATA_GET_ATTR_BY_ID(attr, SAI_MACSEC_SC_ATTR_ENCRYPTION_ENABLE, attrCount, attrList);
```

Comment:
why this is changed? even if wrong, this code is not used, and did you tested this whether this will work ?

Reply:
@Pterosaur, can you look at those changes ?

Reply:
There was compilation error due to this macro hence i corrected it.

Reply:
this function is called loadMACsecAttrFromMACsecSA, so it's getting "SA" attributes, but you changed this to "SC" attribute, how come this was compiling before ?

Reply:
those are 2 different object types, and this is SA object, and now you query SC attribute

Reply:
Commented the code and added SWSS_LOG_THROW to generate exception for author to fix it in future as per discussion.

Reply:
sure, just add comments for other 2 attributes you changed